rechunker/api.py

```diff
@@ -314,13 +374,21 @@ def _rechunk_array(
     source_read = dsa.from_zarr(
         source_array, chunks=read_chunks, storage_options=source_storage_options
     )
+    if source_slice is not None:
+        source_read = source_read[source_slice]
 
     # create target
     shape = tuple(int(x) for x in shape)  # ensure python ints for serialization
     target_chunks = tuple(int(x) for x in target_chunks)
     int_chunks = tuple(int(x) for x in int_chunks)
     write_chunks = tuple(int(x) for x in write_chunks)
 
+    if target_append:
+        target_store_or_group_original = target_store_or_group
+        target_store_or_group = "to_append_" + target_store_or_group
```

Comment:
I don't think this will work on stores that are not paths. Note that the store can also be a mutable mapping (e.g. from s3fs). This is not covered by tests but should be!

Reply:
Yes, this is what I meant by "limited implementation". I will work on that later.